lte/gateway/python/magma/mobilityd/ip_address_man.py

```diff
@@ -153,19 +159,25 @@ def __init__(self,
 
         if self.allocator_type == MobilityD.IP_POOL:
             self._dhcp_gw_info.read_default_gw()
-            self.ip_allocator = IpAllocatorStatic(self._assigned_ip_blocks,
-                                                  self.ip_state_map,
-                                                  self.sid_ips_map)
+            ip_allocator = IpAllocatorPool(self._assigned_ip_blocks,
+                                           self.ip_state_map,
+                                           self.sid_ips_map)
         elif self.allocator_type == MobilityD.DHCP:
             dhcp_store = store.MacToIP()  # mac => DHCP_State
             iface = config.get('dhcp_iface', 'dhcp0')
             retry_limit = config.get('retry_limit', 300)
-            self.ip_allocator = IPAllocatorDHCP(self._assigned_ip_blocks,
-                                                self.ip_state_map,
-                                                iface=iface,
-                                                retry_limit=retry_limit,
-                                                dhcp_store=dhcp_store,
-                                                gw_info=self._dhcp_gw_info)
+            ip_allocator = IPAllocatorDHCP(self._assigned_ip_blocks,
+                                           self.ip_state_map,
+                                           iface=iface,
+                                           retry_limit=retry_limit,
+                                           dhcp_store=dhcp_store,
+                                           gw_info=self._dhcp_gw_info)
+
+        if self.static_ip_enabled:
```

Comment:
I feel this is a little messy since we are instantiating an ip_allocator already given the type of it, and then we're injecting it as a dependency to another class that seems to wrap the logic in it. 

Can we include this wrapping extension into the already defined classes? Or maybe rename the IpAllocatorStatic to something like IPAllocatorStaticWrapper since it looks more like an extension of the ip allocator with subscriberdb client rather than another type of IP allocator.

Reply:
In V1 extended IPAddressManager to check for static IP, so I do not want to extend existing IP allocator classes, given that would duplicate the logic. I can rename the class.